### PR TITLE
boards: up_squared: fixes build errors and include it in sanitycheck

### PR DESCRIPTION
**Pull request by dcpleung**
Due to earlier DTS changes, the names of GPIO macros have also changed. So update the necessary bits to refer to the new naming. Also updated the sample.yaml so that GPIO sample will be built by sanitycheck.

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: drivers/gpio/gpio_intel_apl.c [hunk 1]
```diff
@@ -430,40 +430,41 @@ static const struct gpio_intel_apl_config gpio_intel_apl_cfg = {
 	.islands = {
 		{
 			/* North island */
-			.reg_base = INTEL_APL_GPIO_0_BASE_ADDRESS_0,
+			.reg_base = INTEL_APL_GPIO_D0C50000_BASE_ADDRESS_0,
```
Comment (galak) [inline drivers/gpio/gpio_intel_apl.c]: There should be a dts_fixup.h file that maps what is generated to something this driver would use.   Should be something like CONFIG_APL_GPIO_x_BASE_ADDR (where '_x_' would be something if there is more than one port or block of GPIOs).

**Review by galak — CHANGES_REQUESTED** [review]
Introduce or use the dts_fixup.h for the SoC

File: drivers/gpio/gpio_intel_apl.c (hunk 1 above)
Comment (dcpleung) [reply drivers/gpio/gpio_intel_apl.c]: Added these into the SoC's existing dts_fixup.h

File: soc/x86/apollo_lake/dts_fixup.h [hunk 2]
```diff
@@ -16,4 +16,17 @@
 
 #define CONFIG_IOAPIC_BASE_ADDRESS              INTEL_IOAPIC_FEC00000_BASE_ADDRESS
 
+#define APL_GPIO_BASE_ADDRESS_0			INTEL_APL_GPIO_D0C50000_BASE_ADDRESS_0
```
Comment (galak) [inline soc/x86/apollo_lake/dts_fixup.h]: CONFIG_APL_GPIO..

**Review by galak — DISMISSED** [review]
We try and keep with the convention that dts_fixup.h.  So intead of

```
#define APL_GPIO_BASE_ADDRESS_1			INTEL_APL_GPIO_D0C50000_BASE_ADDRESS_1
```

make it:
```
#define CONFIG_APL_GPIO_BASE_ADDRESS_1			INTEL_APL_GPIO_D0C50000_BASE_ADDRESS_1
```

**Review by galak — APPROVED** [review]

**Pull request merged**